pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -352,13 +360,12 @@ func (cb *ClusterBuilder) buildInboundClusterForPortOrUDS(clusterPort int, bind
 	if cfg != nil {
 		destinationRule := cfg.Spec.(*networking.DestinationRule)
 		if destinationRule.TrafficPolicy != nil {
-			connectionPool, _, _, _ := selectTrafficPolicyComponents(MergeTrafficPolicy(nil, destinationRule.TrafficPolicy, instance.ServicePort))
-			// only connection pool settings make sense on the inbound path.
-			// upstream TLS settings/outlier detection/load balancer don't apply here.
-			cb.applyConnectionPool(cb.push.Mesh, localCluster, connectionPool)
+			opts.policy = MergeTrafficPolicy(opts.policy, destinationRule.TrafficPolicy, instance.ServicePort)
```

Comment:
this doesn't seem right, we should only apply applyConnectionPool not all policy?

Reply:
Actually for inbound cluster, the `applyTrafficPolicy` equals `applyConnectionPool`